internal/c/libqb/src/mac-mouse-support.cpp

```diff
@@ -0,0 +1,74 @@
+// Mouse support functions for macOS
+// These are required to overcome the limitations of GLUT
+
+#include "libqb-common.h"
+
+#include "mac-mouse-support.h"
+
+#include <ApplicationServices/ApplicationServices.h>
+#include <GLUT/glut.h>
+#include <unistd.h>
+
+#define QB64_EVENT_CLOSE 1
+#define QB64_EVENT_KEY 2
+#define QB64_EVENT_RELATIVE_MOUSE_MOVEMENT 3
+#define QB64_EVENT_FILE_DROP 4
+
+extern "C" int qb64_custom_event(int event, int v1, int v2, int v3, int v4, int v5, int v6, int v7, int v8, void *p1, void *p2);
+void GLUT_MOUSEWHEEL_FUNC(int wheel, int direction, int x, int y);
+
+extern int g_MouseX, g_MouseY;
+
+static CFMachPortRef g_EventTap = nullptr;
+static CFRunLoopSourceRef g_RunLoopSource = nullptr;
+
+static CGEventRef macMouseCallback(CGEventTapProxy proxy, CGEventType type, CGEventRef event, void *userInfo) {
```

Comment:
I assume the callback gets called from the same main thread that GLUT runs on (since that should be running the event loop)?

Reply:
Yes. None of this stuff depends on GLUT as it bypasses it and goes straight to the OS.